Modules/overlapped.c

```diff
@@ -2045,14 +2011,15 @@ overlapped_exec(PyObject *module)
         return -1;
     }
 
-    OverlappedState *st = overlapped_get_state(module);
-    st->overlapped_type = (PyTypeObject *)PyType_FromModuleAndSpec(
+    PyTypeObject *overlapped_type = (PyTypeObject *)PyType_FromModuleAndSpec(
```

Comment:
All the `PyType_From*` APIs call `PyType_FromMetaclass`. The recent changes to the `PyType_From*` APIs should not result in a lot of extra work. If that is the case, `PyType_FromMetaclass` should be optimised, if possible.

Reply:
I was under the assumption that the ``PyType_From*`` APIs are used to create a new type object from a spec, not fetch an existing type. It may even leak memory . ``PyType_FromMetaclass`` calls ``PyMem_Malloc`` and eventually assigns it to ``type->tp_name`` eventually without a check.

Reply:
Each call allocates a whole new type on the heap memory.

Reply:
IMO, if there are issues with PEP-489 and the `PyType_From*` APIs they should be addressed in a separate issue and discussed there. Suggesting to mark this conversation as resolved and continue this discussion in an appropriate issue if needed.